### PR TITLE
Support for RED streams.

### PR DESCRIPTION
**Pull request by bgrozev**

### CODE REVIEW

File: jvb/src/main/kotlin/org/jitsi/videobridge/util/PayloadTypeUtil.kt [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            ext.parameters.forEach { parameter ->
+                // In SDP, format parameters don't necessarily come in name=value pairs (see e.g. the format used in
+                // RFC2198). However XEP-0167 requires a name and a value. Our SDP-to-Jingle implementation in
+                // lib-jitsi-meet translates a non-name=value SDP string into a parameter extension with a value but no
+                // name. Here we'll just ignore such parameters, because we don't currently support any and changing the
+                // implementation would be inconvenient (we store them mapped by name).
+                if (parameter.name != null) {
+                    parameters[parameter.name] = parameter.value
+                } else {
+                    logger.warn("Ignoring a format parameter with no name: " + parameter.toXML())
```
Comment (JonathanLennox) [inline jvb/src/main/kotlin/org/jitsi/videobridge/util/PayloadTypeUtil.kt:78]: If this is something that happens frequently will this warning get too chatty?

Comment (bgrozev) [reply jvb/src/main/kotlin/org/jitsi/videobridge/util/PayloadTypeUtil.kt:78]: No, it never happens with an unmodified jicofo. I only noticed it while looking at adding fmtps for RED, which I ended up not committing.

File: jvb/src/main/kotlin/org/jitsi/videobridge/util/PayloadTypeUtil.kt [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                }
+            }
+
+            val rtcpFeedbackSet: MutableSet<String> = CopyOnWriteArraySet()
+            ext.rtcpFeedbackTypeList.forEach { rtcpExtension ->
+                var feedbackType = rtcpExtension.feedbackType
+                val feedbackSubtype = rtcpExtension.feedbackSubtype
+                if (feedbackSubtype != null && feedbackSubtype != "") {
```
Comment (bbaldino) [inline jvb/src/main/kotlin/org/jitsi/videobridge/util/PayloadTypeUtil.kt:86]: There's `isNullOrEmpty` that you can call on a `String?`

Comment (bbaldino) [reply jvb/src/main/kotlin/org/jitsi/videobridge/util/PayloadTypeUtil.kt:86]: Could also leverage `map`/`toCollection` here
```
ext.rtcpFeedbackTypeList.map { rtcpExt ->
    buildString {
      append(rtcpExt.feedbackType)
      if (!rtcpExt.feedbackSubtype.isNullOrEmpty()) {
          append(" ${rtcpExt.feedbackSubtype")
      }
    }
}.toCollection(rtcpFeedbackSet)
```

I wish there was an "append if not null or empty", but don't think one exists in stdlib from what I can tell.

**Comment by bgrozev** [comment]
> Does it still need to be CopyOnWrite?

I don't understand why `CopyOnWrite` was used, but if it needs to be used it shoud be enforced from the class. I pushed a change to JMT to always use `CopyOnWriteArraySet`

**Review by JonathanLennox — APPROVED** [review]

**Pull request merged**